nntrainer/graph/network_graph.cpp

```diff
@@ -214,16 +216,18 @@ int NetworkGraph::realizeMultiOutputType(
  * @fixme: the implementation assumes loss layer should always be at the last
  * layer and the there is only one loss, this assumption is not true
  */
-int NetworkGraph::addLossLayer(const LossType loss_type) {
+int NetworkGraph::addLossLayer(const std::string &loss_type) {
   int status = ML_ERROR_NONE;
   auto const &last_node = LNODE(graph.getSortedNode(graph.size() - 1));
   auto last_layer_node = getSortedLayerNode(graph.size() - 1);
 
   if (last_layer_node->requireLabel()) {
+    if (last_layer_node->getType() != loss_type)
```

Comment:
istrequal here?

Reply:
Case handling for cross entropy might be needed here.

Reply:
> Case handling for cross entropy might be needed here.

You mean for cross vs cross_softmax/sigmoid? 
In the current code, the existing loss layer is a layer added by the user to the model in model description or the API. Below are some scenarios:
1. Already added = cross, loss_type = cross - PASS
2. Already added = cross_sigmoid, loss_type = cross - FAIL as the behavior of the two scenrios become different (in the former case, last layer before loss can be any layer, but for latter case, the previous layer must be sigmoid/softmax).

Reply:
> istrequal here?

done.

Reply:
Thanks for the explanation.

Even for case 1. Isn't it invalid parameter?

since added cross but trying to add another cross after it, something like below?

```cpp
if(last_layer.requireLabel()) {
 return loss_string.empty() ? OK : ERROR
}
```

Reply:
I understand now. You prefer treating setting loss twice as an error.
This is being handled with @lhs8928 when he support multiple losses.
This PR just provides a temporary handling for it.

Reply:
Cool :)